include/zephyr/drivers/i3c.h

```diff
@@ -1111,24 +1320,167 @@ static inline int i3c_recover_bus(const struct device *dev)
  * been updated with SETNEWDA.
  * This method is optional.
  *
+ * @warning
+ * Use cases involving multiple writers to the i3c/i2c devices must prevent
+ * concurrent write operations, either by preventing all writers from
+ * being preempted or by using a mutex to govern writes to the i3c/i2c devices.
+ *
  * @param target Pointer to the target device descriptor
  * @param old_dyn_addr The old dynamic address of target device, 0 if
  *            there was no old dynamic address
  *
  * @retval 0 If successful.
  * @retval -EINVAL If address is not available
  */
-static inline int i3c_reattach_i3c_device(struct i3c_device_desc *target,
-			uint8_t old_dyn_addr)
+static inline int i3c_reattach_i3c_device(struct i3c_device_desc *target, uint8_t old_dyn_addr)
 {
-	const struct i3c_driver_api *api =
-		(const struct i3c_driver_api *)target->bus->api;
+	struct i3c_driver_data *data = (struct i3c_driver_data *)target->bus->data;
+	const struct i3c_driver_api *api = (const struct i3c_driver_api *)target->bus->api;
+	int status = 0;
```

Comment:
duplicate comment?

Reply:
Probably because my internet was down momentarily after I pressed "add comment"... and it allowed me to press the button again. So I did and we have a dupe.